samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -418,12 +420,23 @@ object SamzaContainer extends Logging {
       .orNull
     info("Got checkpoint manager: %s" format checkpointManager)
 
+    val metadataStoreFactory = Option(config.getStartpointMetadataStoreFactory)
+      .map(Util.getObj(_, classOf[MetadataStoreFactory]))
+      .orNull
+    var startpointManager = None: Option[StartpointManager]
+    try {
+      startpointManager = Some(StartpointManager.getWithMetadataStore(metadataStoreFactory, config, samzaContainerMetrics.registry))
+    } catch {
+      case e: Throwable =>
```

Comment:
Minor: maybe catch Exception instead? If it's a Throwable, it's probably pretty bad, so maybe let that exception get thrown.